compose/rest/record.go

```diff
@@ -167,6 +180,108 @@ func (ctrl *Record) Upload(ctx context.Context, r *request.RecordUpload) (interf
 	return makeAttachmentPayload(ctx, a, err)
 }
 
+func (ctrl *Record) ImportInit(ctx context.Context, r *request.RecordImportInit) (interface{}, error) {
+	var (
+		err           error
+		recordDecoder Decoder
+	)
+
+	// Access control.
+	if _, err = ctrl.module.With(ctx).FindByID(r.NamespaceID, r.ModuleID); err != nil {
+		return nil, err
+	}
+
+	f, err := r.Import.Open()
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	// @todo...
+	ext := "json"
+
+	// determine decoder
+	switch strings.ToLower(ext) {
+	case "json", "jsonl", "ldjson", "ndjson":
+		recordDecoder = decoder.NewStructuredDecoder(json.NewDecoder(f))
+
+	case "csv":
+		recordDecoder = decoder.NewFlatReader(csv.NewReader(f))
+
+	// @todo...
+	case "xlss":
+		// recordDecoder = encoder.NewExcelizeDecoder()
+
+	default:
+		return nil, errors.New(fmt.Sprintf("unsupported format (\"%s\")", ext))
+
+	}
+
+	// Make session
+	ses := types.RecordImportSession{
+		SessionID:    factory.Sonyflake.NextID(),
+		ImportFields: recordDecoder.Header(),
+		NamespaceID:  r.NamespaceID,
+		ModuleID:     r.ModuleID,
+	}
+
+	ctrl.session.SetByID(ses.SessionID, &recordImport{decoder: recordDecoder, session: ses})
```

Comment:
1. Move the responsibility of generating session-IDs out of controller into service
2. passing non-exported structs and accept it as interface{}, not something I would be proud of. Pass params in as directly as possible (along with decoder)